application/common/components/MfaBackendTotp.php

```diff
@@ -35,6 +36,44 @@ class MfaBackendTotp extends Component implements MfaBackendInterface
      */
     public MfaApiClient $client;
 
+    /**
+     * Verify response from user is correct for the MFA backend device
+     * @param int $mfaId The MFA ID
+     * @param string $value Value provided by user, such as TOTP number or WebAuthn challenge response
+     * @param string $rpOrigin
+     * @param string $verifyType Only used for WebAuthn
+     * @return bool
+     * @throws BadRequestHttpException
+     * @throws NotFoundHttpException
+     * @throws ServerErrorHttpException
+     * @throws \Exception
+     */
+    public function verify(int $mfaId, string $value, string $rpOrigin = '', string $verifyType = ''): bool
```

Comment:
If you don't mind moving this back down (it was somehow accidentally moved above `init()`), that would make the diff easier to read.